ngraph/frontend/paddlepaddle/src/op/elementwise_ops.cpp

```diff
@@ -25,59 +26,54 @@ NamedOutputs elementwise_ops(const NodeContext& node) {
     if ((axis == -1) || (axis == x_rank - 1) || (x_rank == y_rank)) {
         return node.default_single_output_mapping({std::make_shared<T>(x, y)}, {"Out"});
     } else {
-        // This broadcast can be implemented by either ngraph::Reshape or
-        // ngraph::Broadcast. Since PDPD implicates y_shape is a subsequence of
-        // x_shape starting from axis, to use ngraph::Reshape like Paddle2ONNX,
-        // which is more friendly to PnP.
-        auto broadcast_shape = std::vector<int64_t>(x_rank, 1);
-        PartialShape y_shape = y.get_partial_shape();
-        int32_t i = 0;
-        for (auto it = y_shape.begin(); it != y_shape.end(); ++i, ++it)
-            broadcast_shape[axis + i] = (*it).get_length();
-
-        auto reshape_node = ngraph::opset6::Constant::create(ngraph::element::i64,
-                                                             ngraph::Shape{broadcast_shape.size()},
-                                                             broadcast_shape);
-        auto y_node = std::make_shared<ngraph::opset6::Reshape>(y, reshape_node, false);
+        std::vector<int64_t> indices;
+        for (int64_t i = 0; i < axis; i++)
+            indices.push_back(i);
+        for (int64_t i = y_rank + axis; i < x_rank; i++)
+            indices.push_back(i);
+
+        auto indices_node =
+            default_opset::Constant::create(ngraph::element::i64, ngraph::Shape{indices.size()}, indices);
+        auto y_node = std::make_shared<default_opset::Unsqueeze>(y, indices_node);
         return node.default_single_output_mapping({std::make_shared<T>(x, y_node)}, {"Out"});
```

Comment:
why can't we set auto broadcast to PDPD mode and skip this Unsqueeze? Will it be supported by plugins and further tools? I see this as proper way to read eltwise ops

Reply:
Hi @jane-intel , updated by using PDPD broadcast. also find some bugs in transforming while testing. please check

Reply:
Hi @jane-intel , I also find a bug in transformer if we apply PDPD autobroadcast. There is a workaround to skip this issue. The workaround is in https://github.com/meiyang-intel/openvino/tree/meiyang/op_sequence_fusion_rm_pdpd_broadcast . But this WA will reduce the performance some paddle models. And the fix to apply broadcast to op sequence fusion is in progress.
But after second consideration, these fix for transformer may not only impact paddle. Also the 2022.1 is upcoming, so I don't want to import these changes to 2022.1 to prevent from unknown issues.
So can I revert the commit which is applied the PDPD broadcast and using 'unsqueeze' in 2022.1?

Reply:
Okay, proceed with unsqueeze. I was simply trying to understand if pdpd broadcast for eltwise ops could be a solution here.